text/0000-light-dom.md

```diff
@@ -0,0 +1,218 @@
+ */
+export default class ChildElement extends MacroElement {
+  renderedCallback() {
+    this.querySelector("button").addEventListener("click", this.handleClick);
+  }
+
+  handleClick() {
+    this.dispatchEvent(new CustomEvent("custom", { detail: { macro: true } }));
+  }
+}
+```
+
+### Security (WIP)
+
+- In some applications, light-dom components may not be allowed... it’s up to the app context
+  - **what is the behavior when it’s not allowed?**
+  - Some applications might disable light-dom as a whole
+  - Some applications might disable light-dom selectively using a “privileged code” model
+
+### Component Migration
+
+There is no migration of the existing components. Existing components inherit from `LightningElement` and light DOM components will inherit from `MacroElement` and they are two different things. If a component author wishes to use Light DOM for any of their existing components, they will have to make the relevant changes manually.
+
+### Server Side Rendering
+
+The engine-server module should provide the SSR capability to seamlessly render Shadow DOM or Light DOM. It should include the component children, as well as generated styles.
+
+## Adoption strategy
+
+This new feature does not break any existing components, it simply adds a new feature that developers have to opt for. There is no migration of the existing components needed.
+
+This feature should be exposed and explained to the component library developers as they might change how they develop their components internally.
+
+## How we teach this
+
+Shadow DOM and Light DOM are already names accepted by the industry, see: [Terminology: light DOM vs. shadow DOM](https://developers.google.com/web/fundamentals/web-components/shadowdom?hl=en).
+We need to provide the proper documentation to educate the LWC developers:
+
+- What are the differences between Shadow DOM and Light DOM
+- We need a guide on when to use one or the other
+
+## Open questions
+
+**How to deal with different namespaced packages claiming the same custom element name?** 
+
+As of today on the Salesforce platform, all the LWC components are authored under the `c` namespace. This namespace dictates the prefix for of the custom elements rendered in the DOM. When delivered via a namespaced package, the components are referenced using their actual namespace (eg. `lightning`, `my_ns`, etc.). However internal component references in the package is done using the `c` namespace. 
+
+Because of this if 2 manage package publish a `button` component, the 2 components will be rendered as `c-button` in DOM. This is problematic because a custom element name can only be bound to a single element constructor. Even if this is not a new issue, the [scoped custom element registry proposal](https://github.com/WICG/webcomponents/issues/716) would solve this issue for shadow DOM components. This is still an issue for light DOM component as scoped custom elements registry can only be attached to a shadow root.  
```

Comment:
Is this a LEX-specific issue, or can it be abstracted to apps not using LEX? Or is it more in the wheelhouse of Locker?

Reply:
This is a Salesforce platform-specific issue. We would run into this issue in LEX and in Commerce Cloud as well. Even if the Locker team have an opinion, it certainly something the LWC team has to drive.